AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Session/SessionTracker.swift

```diff
@@ -5,12 +5,158 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+import Amplify
 import Foundation
+import UIKit
 
 class SessionTracker: InternalPinpointClient {
+    private var session: PinpointSession? = nil
+    private var backgroundTask: UIBackgroundTaskIdentifier = .invalid
```

Comment:
we'll need abstractions around all these UIKit-related symbols (`UIBackgroundTaskIdentifier`, `UIApplication`) to support our cross-platforms effort.
I'd recommend having protocols and add conformance using Swift compiler `canImport` directive.

Reply:
in this case you do not need an abstraction because there is no equivalent outside of UIKit.

Reply:
I will extract this behaviour to a separate class, so we keep UIKit out of this one.